src/state-model.test.js

```diff
@@ -0,0 +1,164 @@
+import assert from 'assert';
+import sinon from 'sinon';
+
+import _ from 'underscore';
+import { Model } from 'backbone';
+
+import getStateMixin from './state-model.js';
+
+// Methods that we expect the mixin *always* to have, regardless of
+// customizations.
+var expectedInterface = {
+    bindStateEvents: sinon.match.func,
+    broadcastStateEvents: sinon.match.func
+};
+
+describe('getStateMixin', function() {
+    var mixin, State;
+
+    describe('by default', function() {
+        beforeEach(function() {
+            mixin = getStateMixin();
+            State = Model.extend(mixin);
+        });
+
+        it('returns a mixin with three properties', function() {
```

Comment:
isn't this an implementation detail? (ditto for the next two tests)

Reply:
Kind of. These things also somewhat matter to the end user, but it certainly is a white box test. I often test for (semi) implementation details because they provide sanity checks for later tests.